cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/preview/DefaultPreviewManager.java

```diff
@@ -0,0 +1,148 @@
+    ArtifactSummary artifactSummary = appRequest.getArtifact();
+    Id.Namespace artifactNamespace =
+      ArtifactScope.SYSTEM.equals(artifactSummary.getScope()) ? Id.Namespace.SYSTEM : namespaceId.toId();
+    Id.Artifact artifactId =
+      Id.Artifact.from(artifactNamespace, artifactSummary.getName(), artifactSummary.getVersion());
+
+    // if we don't null check, it gets serialized to "null"
+    String configString = appRequest.getConfig() == null ? null : GSON.toJson(appRequest.getConfig());
+    try {
+      applicationLifecycleService.deployApp(namespaceId.toId(), previewId.getPreview(), artifactId, configString,
+                                            new ProgramTerminator() {
+                                              @Override
+                                              public void stop(Id.Program programId) throws Exception {
+
+                                              }
+                                            });
+    } catch (Exception e) {
+      previewStatusMap.put(previewId, new PreviewStatus(PreviewStatus.Status.DEPLOY_FAILED,
+                                                        Throwables.getRootCause(e).getMessage()));
+      throw Throwables.propagate(e);
+    }
+
+    // Assume that we are starting the SmartWorkflow in the DatapipelineApp
+    ProgramId programId = new ProgramId(namespaceId.getNamespace(), previewId.getPreview(), ProgramType.WORKFLOW,
+                                        "DataPipelineWorkflow");
+    ProgramRuntimeService.RuntimeInfo runtimeInfo = programLifecycleService.start(programId,
+                                                                                  new HashMap<String, String>(),
+                                                                                  new HashMap<String, String>(), false);
+
+    runtimeInfo.getController().addListener(new AbstractListener() {
+      @Override
+      public void init(ProgramController.State currentState, @Nullable Throwable cause) {
+        previewStatusMap.put(previewId, new PreviewStatus(PreviewStatus.Status.RUNNING, null));
+      }
+
+      @Override
+      public void completed() {
+        previewStatusMap.put(previewId, new PreviewStatus(PreviewStatus.Status.COMPLETED, null));
+      }
+
+      @Override
+      public void error(Throwable cause) {
+        previewStatusMap.put(previewId, new PreviewStatus(PreviewStatus.Status.RUN_FAILED,
+                                                          Throwables.getRootCause(cause).getMessage()));
+      }
+    }, Threads.SAME_THREAD_EXECUTOR);
+
+    return previewId;
+  }
+
+  @Override
+  public PreviewStatus getStatus(PreviewId previewId) throws NotFoundException {
+    if (!previewStatusMap.containsKey(previewId)) {
+      throw new NotFoundException(previewId + " not found");
+    }
+    return previewStatusMap.get(previewId);
+  }
+
+  @Override
+  public Map<String, Map<String, List<Object>>> getData(PreviewId previewId) {
+    return null;
```

Comment:
Why return null? Even it's to be implemented, it's better to adhere to the contract (non-nullable in this case), hence better return an empty map. Same for the other methods.